README.md

```diff
@@ -1,76 +1,130 @@
+
+# Getting started
+
+1. Clone the repository:
+
+    ~~~
+    $ git clone https://github.com/swcarpentry/amy.git
+    $ cd amy
+    ~~~
+
+2. Install Django and its dependencies:
 
     ~~~
     $ sudo python3 -m pip install -r requirements.txt
     ~~~
 
-2. Install [Bower](http://bower.io/) - the tool that manages Amy's JavaScript and CSS dependencies:
+    If you're experienced Python programmer, feel free to create a Python3-compatible virtualenv for Amy and install dependencies from `requirements.txt`.
+
+3. Install [Bower](http://bower.io/) — the tool that manages Amy's JavaScript and CSS dependencies:
 
     ~~~
-    $ npm install -g bower
+    $ sudo npm install -g bower
     ~~~
 
-3.  Setup your local database.  There are two ways you can do this.
+    You may need some additional dependencies to install [Bower](http://bower.io/), such as [NodeJS](https://nodejs.org/) and [npm](https://www.npmjs.com/).
+
+4. Setup your local database.  There are two ways you can do this:
 
     1. If you have access to the legacy data:
 
         1. Create an empty database by running:
 
-           ~~~
-           $ make migrations
-           ~~~
+            ~~~
+            $ make migrations
+            ~~~
 
         2. Fill that database by running:
 
-           ~~~
-           $ make import
-           ~~~
+            ~~~
+            $ make import
+            ~~~
 
-    2. Otherwise set up the redacted data with:
+    2. Otherwise set up the redacted (development-ready) data with:
 
-       ~~~
-       $ make database
-       ~~~
+        ~~~
+        $ make database
+        ~~~
 
-4.  Create a administrator account.
+5. Create an administrator account:
 
     ~~~
     $ python3 manage.py createsuperuser
     ~~~
 
-5.  Start a local Django development server by running (this also installs front-end dependencies for Amy):
+6. Start a local Django development server by running:
 
     ~~~
     $ make serve
     ~~~
 
-6.  Open [http://localhost:8000/workshops/](http://localhost:8000/workshops/) in your browser and start clicking.
+    **Note**:  this also installs front-end dependencies for Amy, such as jQuery or Bootstrap.
+
+7. Open [http://localhost:8000/workshops/](http://localhost:8000/workshops/) in your browser and start clicking.
 
     Use the administrator account that you created.
 
-**Note**: please [check with us](mailto:gvwilson@software-carpentry.org)
-or open a [discussion issue](https://github.com/swcarpentry/amy/labels/discussion)
+# Upgrading
+
+1. Update the code:
+
+    1. Get the list of changes:
+
+        ~~~
+        $ git fetch
+        ~~~
+
+    2. Look for the newest tag:
+
+        ~~~~
+        $ git tag -n
+        ~~~~
+
+    3. Get the code from the newest tag:
+
+        ~~~~
+        $ git checkout tags/<tag_name>
+        ~~~~
+
+2. Update dependencies:
+
+    ~~~
+    $ sudo python3 -m pip install --upgrade -r requirements.txt
+    $ bower update
```

Comment:
On Tue, May 12, 2015 at 12:50:16PM -0700, Piotr Banaszkiewicz wrote:
> > +    $ bower update
> 
> We're using `bower install` in `Makefile:77`. Will it update
> packages if they're outdated?

It will if you've bumped a dependency in your bower.json.  The only
difference will be if:

a. You have an existing install that already satisfies the bower.json
   dependency, and
b. There have been subsequent releases of that package that would also
   satisfy the bower.json dependency.

In that case, I see no need to update the local package.  If you do
have a reason to update the local package (e.g. there was a bug-fix in
a dependency), you should be bumping the bower.json listing.

If you periodically want to upgrade to current releases (e.g. because
you just assume upstream is fixing bugs that you don't know about),
then an update call is fine, but you can do that by hand.  And a move
like that would be part of amy development (you're forward-porting amy
to new dependencies), not part of a standard amy upgrade.